bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Button.java

```diff
@@ -1107,7 +1107,24 @@ LRESULT wmColorChild (long wParam, long lParam) {
 		// Button has "transparent" portions which need to be filled with
 		// parent's (and not Button's) background. For example, SWT.PUSH
 		// button ~2px transparent area around the button.
-		return parent.wmColorChild(wParam, lParam);
+
+		// Issue 848: wmColorChild() contains a hack that paints entire
```

Comment:
Do you mean org.eclipse.swt.widgets.Control.wmColorChild(long, long)?
```suggestion
		// Issue 848: Control.wmColorChild.() contains a hack that paints entire
```
i don't see a recent change there and wonder why the error is a recent regression then. Also the bug report is about light theme, so i wonder why THEME_BACKGROUND does influence light theme.

Reply:
Comment improved.

The regression happened because I used `parent.wmColorChild()` (right here in the code you're looking at), which contains the unexpected hack.

Reply:
Light theme is important because `Control.findBackgroundControl()` has condition `background != -1` which prevents `THEME_BACKGROUND` hack.
So it's not about light theme, but about NOT setting any colors.